### DISCUSSION TITLE
How to use video URL for thumbnail

### DISCUSSION THREAD

**Discussion by mehroozkhan** [kean/Nuke#498] [Q&A]
Hi @kean , 

is there any way to use video url with Nuke , so it can display thumbnail image of video, and cache it as well.
actually i am using your library in a table view, and if i use my own code to generate thumbnail from video url then there are UI glitches when user scroll and video thumbnail is not generated yet and that cell is reused to display image with Nuke. 

Any suggestions ?

Thanks in advance.

**Comment by kean** [comment]
You can check NukeUI for an example https://github.com/kean/NukeUI/blob/main/Sources/VideoImageDecoder.swift. It generates thumbnails before displaying a video. The system downloads the entire video though - I don't know if that's acceptable to you.

**Reply by kean** [reply]
Please try the latest version. There've been a number of changes to video support added since then. Preview generation is definitely one of them. And you don't need NukeUI anymore, most of the video support is built directly into the main framework.

**Comment by realmtai** [comment]
The above never worked for me. So I ended up using the AVKit to generate a videoThumbnail instead. Hope this help someone out in the future.

``` swift
    static func videoThumbnail(from url: URL, at time: CMTime = .zero, width: CGFloat = 300) async -> UIImage? {
        let asset = AVURLAsset(url: url)
        let imageGenerator = AVAssetImageGenerator(asset: asset)
        imageGenerator.appliesPreferredTrackTransform = true
        imageGenerator.maximumSize = CGSize(width: width, height: 0)

        do {
            // Use the async image(at:) method for a more modern approach.
            let (cgImage, _) = try await imageGenerator.image(at: time)
            return UIImage(cgImage: cgImage)
        } catch {
            return nil
        }
    }

```
